EasyClangComplete.py

```diff
@@ -164,8 +171,26 @@ def on_close(self, view):
                 return
             self.completer.remove(view.buffer_id())
 
+    def completion_finished(self, future):
+        """ Callback called when completion async function has returned. Checks
+        if job id equals the one that is expected now and updates the
+        completion list that is going to be used in on_query_completions
+
+        Args:
+            future (concurrent.Future): future holding completion result
+        """
+        if future.done():
+            (job_id, completions) = future.result()
+            if job_id == self.current_job_id:
+                self.current_completions = completions
+                if self.current_completions:
```

Comment:
@rchl this solves the loop while being exactly what we want. If the popup window is shown because at the time this job was generated the `on_query_completions` returned `Tools.SHOW_DEFAULT_COMPLETIONS` - it will still be shown, while if it was not shown it remains that way.